### PR TITLE
Localize times

### LINKED ISSUE #8583: time & date formats are different across Cockpit
This will potentially open a can of worms, since we're a bit inconsistent in general with regards to time formats in Cockpit. It would be good to harmonize those.

My system time on the Server page is displayed like this:
`2018-02-08 12:56`

But on the services page it's displayed like this:
`1/19/2018, 1:57:36 PM`

The first feels more normal to me, since I'm used to the Swedish system, while the second seems to be the US format.

### PR DESCRIPTION
**Pull request by marusak**
Fixes #8583 
I haven't found any other place, where the format would not be localized. If you find something, please point it out, and I'll fix it out.
The only other place related to time, that should be localized, are calendars:
calendarbefore
(names of days, on which day week starts... - this can be done with setting up `language` but it requires to import all locale files - which cannot be automated... But there are  also other options, so I just plan to open issue for that and take a look at it separately).

I am not sure about that change on the overview page. It seems so obvious that I doubt it was not purposely that way? Now looks like this in English and is localized: 
timeformat
Is that right?

Likely breaks tests, so spinning them out

### CODE REVIEW

**Comment by martinpitt** [comment]
> I am not sure about that change on the overview page

In master I see "2020-02-04 16:19" right now, which is ISO format; both in German and in English. I suppose that works in every language, but indeed it seems fine to actually localize it.

File: pkg/systemd/overview-cards/serverTime.js [hunk 1]
```diff
@@ -61,13 +62,10 @@ export function ServerTime() {
...
 
     self.format = function format(and_time) {
-        var string = self.utc_fake_now.toISOString();
-        if (!and_time)
-            return string.split('T')[0];
-        var pos = string.lastIndexOf(':');
-        if (pos !== -1)
-            string = string.substring(0, pos);
-        return string.replace('T', ' ');
+        moment.locale(cockpit.language); // HACK: when done at the top of the file, c.language is undefined
```
Comment (martinpitt) [inline pkg/systemd/overview-cards/serverTime.js]: I think it should rather live in pkg/systemd/overview.jsx . @allisonkarlitskaya needs that for PR #13473 as well.

File: test/verify/check-system-info [hunk 2]
```diff
@@ -252,7 +252,7 @@ class TestSystemInfo(MachineCase):
         b.click("#systime-apply-button")
         b.wait_popdown("system_information_change_systime")
 
-        b.wait_text("#system_information_systime_button", "2020-01-24 08:03")
+        b.wait_text("#system_information_systime_button", "Jan 24, 2020 8:03 AM")
```
Comment (martinpitt) [inline test/verify/check-system-info]: Most probably you need to conditionalize this for 8-2-distropkg.

Comment (marusak) [reply test/verify/check-system-info]: of course it will! I thought about it, but of course totally forgot about it...

**Comment by martinpitt** [comment]
@PabloGrok : You'd need to pick a locale with 24 hour format, perhaps en_GB?

**Comment by PabloGrok** [comment]
But I don't want British idioms in the UI. Shouldn't time formats be decoupled from locales?

**Comment by martinpitt** [comment]
Unfortunately browsers don't have the fine-grained locale configuration like Linux has (`LC_TIME`, `LC_MONETARY`, `LC_MESSAGES` and such).

**Comment by garrett** [comment]
Additionally, the first day of the week is different in various places.

first day of the week
https://en.wikipedia.org/wiki/Week#/media/File:First_Day_of_Week_World_Map.svg

And either we're assuming English means US and have German incorrect, or we (or PF) hardcode the first day of the week as Sunday:

Screenshot 2021-11-08 at 11-47-02 Overview - garrett Bolt

Screenshot 2021-11-08 at 11-47-46 Überblick - garrett Bolt


PatternFly has a `weekStart` option. I guess we're not handling that with locales and/or not passing it to the PF widget:

Name | Type | Default | Description
-- | -- | -- | --
weekStart | 0 \| 1 \| 2 \| 3 \| 4 \| 5 \| 6 \| Weekday |   | Day of week that starts the week. 0 is Sunday, 6 is Saturday.

**Comment by martinpitt** [comment]
@garrett : Indeed -- eww, that sucks. Standard Unix/Linux locale information has had that piece of information forever:
```
❱❱❱ LC_TIME=en_US.UTF-8 locale -k -c LC_TIME | grep first_weekday
first_weekday=1

❱❱❱ LC_TIME=de_DE.UTF-8 locale -k -c LC_TIME | grep first_weekday
first_weekday=2
```

However, I didn't find anything in [Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl) which would expose that piece of information to JS. There is a JS standard issue about this very topic: https://github.com/tc39/ecma402/issues/6 but it has been open for 6 years now (but there is some progress).

So I'm afraid right now the only thing which we can do is to keep a copy of [that information](https://github.com/unicode-cldr/cldr-core/blob/master/supplemental/weekData.json#L59) in our pkg/lib. I'll look into it.

**Comment by martinpitt** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/weekInfo would be that, but it doesn't work in Firefox, Chromium, nor node.js.

**Comment by martinpitt** [comment]
I sent PR 16587 to at least fix this for most users.

**Pull request merged**